scripts/run_bench.py

```diff
@@ -0,0 +1,137 @@
+import argparse
+import asyncio
+import json
+import logging
+import sys
+import time
+import uuid
+from contextlib import asynccontextmanager
+from datetime import timedelta
+from typing import AsyncIterator
+
+from temporalio import activity, workflow
+from temporalio.testing import WorkflowEnvironment
+from temporalio.worker import UnsandboxedWorkflowRunner, Worker
+from temporalio.worker.workflow_sandbox import SandboxedWorkflowRunner
+
+
+@workflow.defn
+class BenchWorkflow:
+    @workflow.run
+    async def run(self, name: str) -> str:
+        return await workflow.execute_activity(
+            bench_activity, name, start_to_close_timeout=timedelta(seconds=30)
+        )
+
+
+@activity.defn
+async def bench_activity(name: str) -> str:
+    return f"Hello, {name}!"
+
+
+async def main():
+    logging.basicConfig(
+        format="%(asctime)s.%(msecs)03d %(levelname)-8s %(message)s",
+        level=logging.WARN,
+        datefmt="%Y-%m-%d %H:%M:%S",
+    )
+
+    logger = logging.getLogger(__name__)
+    max_mem = -1
+
+    parser = argparse.ArgumentParser(description="Run bench")
+    parser.add_argument("--workflow-count", type=int, required=True)
+    parser.add_argument("--sandbox", action=argparse.BooleanOptionalAction)
+    parser.add_argument("--max-cached-workflows", type=int, required=True)
+    parser.add_argument("--max-concurrent", type=int, required=True)
+    args = parser.parse_args()
+
+    @asynccontextmanager
+    async def track_mem() -> AsyncIterator[None]:
+        # We intentionally import in here so the sandbox doesn't grow huge with
+        # this import
+        import psutil
+
+        # Get mem every 800ms
+        process = psutil.Process()
+
+        async def report_mem():
+            nonlocal max_mem
+            while True:
+                try:
+                    await asyncio.sleep(0.8)
+                finally:
+                    # TODO(cretz): "vms" appears more accurate on Windows, but
+                    # rss is more accurate on Linux
+                    used_mem = process.memory_info().rss
+                    if used_mem > max_mem:
+                        max_mem = used_mem
+
+        report_mem_task = asyncio.create_task(report_mem())
+        try:
+            yield None
+        finally:
+            report_mem_task.cancel()
```

Comment:
Is it worth waiting for the task to be cancelled here?

Reply:
Just interrupts a sleep, no value I don't think, but I guess I could